File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/dokumentarkiv/DokumentDto.java
package no.nav.melosys.tjenester.gui.dto.dokumentarkiv;

import java.util.ArrayList;
import java.util.List;

import no.nav.melosys.domain.arkiv.ArkivDokument;
import no.nav.melosys.domain.arkiv.DokumentVariant;

public class DokumentDto {
    public final String dokumentID;
    public final String tittel;
    public final List<String> logiskeVedlegg;
    public final DokumentVariant.Filtype filtype;

    public DokumentDto(String tittel) {
        this.dokumentID = null;
        this.tittel = tittel;
        this.logiskeVedlegg = new ArrayList<>();
        this.filtype = null;
    }

    public DokumentDto(String dokumentID, String tittel, List<String> logiskeVedlegg, DokumentVariant.Filtype filtype) {
        this.dokumentID = dokumentID;
        this.tittel = tittel;
        this.logiskeVedlegg = logiskeVedlegg;
        this.filtype = filtype;
    }

    public static DokumentDto av(ArkivDokument dokument) {
        return new DokumentDto(
            dokument.getDokumentId(),
            dokument.getTittel(),
            dokument.hentLogiskeVedleggTitler(),
            dokument.arkivVariantFiltype()
        );
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/dokumentarkiv/DokumentDto.kt
package no.nav.melosys.tjenester.gui.dto.dokumentarkiv

class DokumentDto {
    @JvmField val dokumentID: String?
    @JvmField val tittel: String
    @JvmField val logiskeVedlegg: List<String>

    constructor(tittel: String) {
        this.dokumentID = null
        this.tittel = tittel
        this.logiskeVedlegg = ArrayList()
    }

    constructor(dokumentID: String?, tittel: String, logiskeVedlegg: List<String>) {
        this.dokumentID = dokumentID
        this.tittel = tittel
        this.logiskeVedlegg = logiskeVedlegg
    }
}
